microceph/ceph/operations.go

```diff
@@ -0,0 +1,284 @@
+package ceph
+
+import (
+	"fmt"
+
+	"github.com/canonical/lxd/shared/logger"
+
+	microCli "github.com/canonical/microcluster/v2/client"
+
+	"github.com/canonical/microceph/microceph/client"
+)
+
+// RunOperations runs the provided operations or prints out the action plan.
+func RunOperations(name string, operations []Operation, dryRun, force bool) error {
+	for _, ops := range operations {
+		if dryRun {
+			fmt.Println(ops.DryRun(name))
+		} else {
+			err := ops.Run(name)
+			if err != nil {
+				logger.Errorf("%v", err)
+				// Skip the error if forced
+				if force {
+					continue
+				}
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Operation is a interface for ceph and microceph operations.
+type Operation interface {
+	// Run executes the operation and return the error if any.
+	Run(string) error
+
+	// DryRun returns the string representation of the operation.
+	DryRun(string) string
+}
+
+// ClusterOps is the base struct for all operations.
+type ClusterOps struct {
+	CephClient    client.ClientInterface
+	ClusterClient *microCli.Client
+}
+
+// CheckNodeInClusterOps is an operation to check if a node is in the microceph cluster.
+type CheckNodeInClusterOps struct {
+	ClusterOps
+}
+
+// Run checks if a node is in the microceph cluster.
+func (o *CheckNodeInClusterOps) Run(name string) error {
```

Comment:
I wonder if its necessary to go via the API client here. At this point we're in server code and could call the resp. handler directly I think. 
Similarly in the below client calls

Reply:
Sorry, I am not sure what do you mean here, can you elaborate more?

> At this point we're in server code

The operations are not run in server side (i.e. `microcephd`)

Reply:
Should be addressed in the recent commits